lib/terrafying/components/letsencrypt.rb

```diff
@@ -146,8 +148,11 @@ def create_keypair_in(ctx, name, options = {})
           dns_names: [],
           ip_addresses: [],
           curve: "P384",
+          zone: ""
         }.merge(options)
 
+        @zones << options[:zone]
```

Comment:
Is this why you end up with empty strings? Is it worth just omitting it from the options hash and then only adding it to zones if it's defined?

Reply:
I was originally only getting empty strings due to the test I'd written, which was just setting 'zone' to be a string, rather than ultimately using the `Terrafying::Components::Zone.find(zone)` method - I think in real world use we will only see the returned zone class, or a nil

Reply:
Ah yeah, sorry I see what you're saying - if we call create_keypair_in without a zone then we are appending an empty (default) string to the zones array each time, so yes, we should only add to zones if the option is defined